cmd/sourced/cmd/compose.go

```diff
@@ -66,18 +68,28 @@ type composeSetDefaultCmd struct {
 	Command `name:"set" short-description:"Set the active docker compose file" long-description:"Set the active docker compose file"`
 
 	Args struct {
-		Version string `positional-arg-name:"version" description:"Either a revision (tag, full sha1) or a URL to a docker-compose.yml file"`
+		Index string `positional-arg-name:"index" description:"Index of the docker compose file returned from 'sourced compose list'"`
 	} `positional-args:"yes" required:"yes"`
 }
 
 func (c *composeSetDefaultCmd) Execute(args []string) error {
-	err := composefile.SetActive(c.Args.Version)
+	files, err := composefile.List()
+
 	if err != nil {
 		return err
 	}
 
-	fmt.Println("Active docker compose file was changed successfully.")
-	fmt.Println("To update your current installation use `sourced restart`")
+	index, err := strconv.Atoi(c.Args.Index)
+
+	if err == nil && index >= 0 && index < len(files) {
+		active := files[index]
+		err = composefile.SetActive(active)
+		fmt.Println("Active docker compose file was changed successfully.")
+		fmt.Println("To update your current installation use `sourced restart`")
```

Comment:
What's happening if the user inits with one config using services `a,b,c` changes to a new compose defining services `c,d`, and then the user restarts?
Isn't it error prone because the potential mismatch between old and new config? Or can we assume it's stable enough for oficial versions?
Should that problem be handled by the user depending on initial and final compose configs?

Reply:
Is it enough to check SetActive error?

Reply:
This is more an open question for @src-d/applications, to consider if we want to handle that edgecase, or if it should be handled by the user running `prune` before changing the active compose.

Reply:
I think this is something to be discussed in the DD about `sourced` upgrade. IMO if the changes in the services occurred because of an upgrade of `sourced` (so something expected), then it should be handled by the upgrade process, or in case that it's not possible, it should be correctly notified to the user. Otherwise, if the change occurred because the user is using a custom compose file, then he/she is an advanced user and I would assume that he/she knows what to do.